src/client/Client.cc

```diff
@@ -6495,6 +6476,44 @@ void Client::tick()
   }
 }
 
+void Client::start_tick_thread()
+{
+  upkeeper = std::thread([this]() {
+    using Ms = std::chrono::milliseconds;
+    using std::chrono::time_point_cast;
+    using time = ceph::coarse_mono_time;
+
+    auto last_tick = time_point_cast<Ms>(time::min());
+
+    std::unique_lock cl(client_lock);
+    while (!tick_thread_stopped) {
+      auto now = time_point_cast<Ms>(clock::now());
+      auto since = now - last_tick;
+      auto d_interval = Ms(0);
+
+      Ms t_interval(static_cast<int64_t>(1000 * g_conf().get_val<double>("client_tick_interval")));
+      // Clear the debug inject tick delay
+      if (unlikely(cct->_conf->client_debug_inject_tick_delay > 0)) {
```

Comment:
Sure.

Reply:
This won't work for most use cases of `libcephfs` excepting the `ceph-fuse`, because the `libcephfs` won't set the `g_ceph_context`. So the `g_conf()` will crash always.

The `ceph-fuse` will call the `global_init()` and it will set the `g_ceph_context`. I am not very sure should we set it for the whole `libcephfs` ? @batrick Any idea ?

Reply:
The simplest way is back to use `cct->_conf->XXX`.

Reply:
Sorry. We should be doing `cct->_conf.get_val<>`.

Reply:
Works, Thanks.